### ISSUE TITLE
Unable to load public repository

### ISSUE THREAD

**Issue by igorlino** [Hacklone/private-bower#135]
Seems that running the service is having problems getting the public packages.

Below is the output from the console. And also, attached the configuration I'm using, which is basically has no change besides that it has a proxy. I don't think is a proxy issue since if I try to access manually http://bower.herokuapp.com/packages/   I get the packages in JSON format.

The error happens because for some reason we get back an HTML website response, instead of the JSON.

```
>> private-bower --config bower-config.json                                                                                      
[2015-04-23 08:59:45.068] [INFO] console - [bower]  23/4/15 08:59:45   Refreshing public packages...                                                                          
[2015-04-23 08:59:45.169] [INFO] console - [bower]  23/4/15 08:59:45   private-bower server started                                                                           
[2015-04-23 08:59:45.420] [INFO] console - [bower]  23/4/15 08:59:45   Could not load public packages, invalid data!                                                          
[2015-04-23 08:59:45.423] [INFO] console - [SyntaxError: Unexpected token <]                                                                                                  
SyntaxError: Unexpected token <                                                                                                                                               
    at Object.parse (native)                                                                                                                                                  
    at _parsePackage (C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\lib\service\packageStores\publicPackageStore.js:28:33)                                
    at processData (C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\lib\service\packageStores\publicPackageStore.js:87:30)                                  
    at C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\lib\service\packageStores\publicPackageStore.js:76:13                                                
    at Object.ConnectManager.handleResponse (C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\node_modules\node-rest-client\lib\node-rest-client.js:431:5)   
    at Object.ConnectManager.handleEnd (C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\node_modules\node-rest-client\lib\node-rest-client.js:404:10)       
    at IncomingMessage.<anonymous> (C:\Users\icampista\AppData\Roaming\npm\node_modules\private-bower\node_modules\node-rest-client\lib\node-rest-client.js:570:13)           
    at IncomingMessage.emit (events.js:129:20)                                                                                                                                
    at _stream_readable.js:908:16                                                                                                                                             
    at process._tickCallback (node.js:355:11)                                                                                                                                 
[2015-04-23 08:59:45.432] [INFO] console - [bower]  23/4/15 08:59:45   data = <!DOCTYPE html>                                                                                 
    <html>                                                                                                                                                                    
    <head>                                                                                                                                                                    
      <title>Heroku | No such app</title>                                                                                                                                     
      <style type='text/css'>                                                                                                                                                 
      body {                                                                                                                                                                  
        background-color: white;                                                                                                                                              
        color: #333333;                                                                                                                                                       
          font-family: Arial, sans-serif;                                                                                                                                     
        margin: 0;                                                                                                                                                            
        padding: 36px;                                                                                                                                                        
        line-height: 1;                                                                                                                                                       
        font-size: 14px; }                                                                                                                                                    
                                                                                                                                                                              
      .section {                                                                                                                                                              
        margin-bottom: 36px; }                                                                                                                                                
        .section.friendly {                                                                                                                                                   
          color: #222222; }                                                                                                                                                   
          .section.friendly h1 {                                                                                                                                              
            font-size: 26px;                                                                                                                                                  
            background-color: #dad8e4;                                                                                                                                        
            padding: 18px 22px 15px 22px;                                                                                                                                     
            margin: 0;                                                                                                                                                        
            overflow: hidden; }                                                                                                                                               
            .section.friendly h1 strong {                                                                                                                                     
              display: inline-block;                                                                                                                                          
              float: left; }                                                                                                                                                  
            .section.friendly h1 small {                                                                                                                                      
              display: inline-block;                                                                                                                                          
              float: right;                                                                                                                                                   
              text-align: right;                                                                                                                                              
              font-size: 18px;                                                                                                                                                
              padding-top: 4px;                                                                                                                                               
              color: #333333; }                                                                                                                                               
          .section.friendly .article {                                                                                                                                        
            border: 4px solid #dad8e4;                                                                                                                                        
            padding: 24px 18px 18px 18px; }                                                                                                                                   
            .section.friendly .article h3 {                                                                                                                                   
              font-size: 20px;                                                                                                                                                
              margin: 0 0 18px 0; }                                                                                                                                           
            .section.friendly .article a {                                                                                                                                    
              color: #6b6ceb; }                                                                                                                                               
              .section.friendly .article a:visited {                                                                                                                          
                color: #1d1d3b; }                                                                                                                                             
            .section.friendly .article p {                                                                                                                                    
              font-size: 14px; }                                                                                                                                              
            .section.friendly .article ul {                                                                                                                                   
              list-style-type: square; }                                                                                                                                      
      .section.original {                                                                                                                                                     
        background-color: #eeeeee;                                                                                                                                            
        color: #444444; }                                                                                                                                                     
        .section.original h2 {                                                                                                                                                
          background-color: #dddddd;                                                                                                                                          
          margin: 0;                                                                                                                                                          
          padding: 18px 22px 18px 22px;                                                                                                                                       
          font-size: 20px; }                                                                                                                                                  
        .section.original pre {                                                                                                                                               
          margin: 0;                                                                                                                                                          
          padding: 18px 22px 18px 22px;                                                                                                                                       
          overflow: auto;                                                                                                                                                     
          font-family: monaco, monospaced; }                                                                                                                                  
          .section.original pre code {                                                                                                                                        
            display: block;                                                                                                                                                   
            font-size: 11px;                                                                                                                                                  
            width: 100%; }                                                                                                                                                    
    </style>                                                                                                                                                                  
      <script type="text/javascript">                                                                                                                                         
var _gaq = _gaq || [];                                                                                                                                                        
_gaq.push(['_setAccount', ', UA-2989055-7']);                                                                                                                                 
_gaq.push(['_setDomainName', '.heroku.com']);                                                                                                                                 
_gaq.push(['_trackPageview']);                                                                                                                                                
(function() {                                                                                                                                                                 
var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;                                                                                      
ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';                                                             
var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);                                                                                         
})();                                                                                                                                                                         
</script>                                                                                                                                                                     
    </head>                                                                                                                                                                   
    <body>                                                                                                                                                                    
        <div class='container'>                                                                                                                                               
          <div class='section friendly'>                                                                                                                                      
            <h1><strong>Heroku | No such app</strong></h1>                                                                                                                    
            <div class='article'>                                                                                                                                             
              <p>There is no app configured at that hostname.<br/>Perhaps the app owner has renamed it, or you mistyped the URL.</p>                                          
            </div>                                                                                                                                                            
          </div>                                                                                                                                                              
        </div>                                                                                                                                                                
    </body>                                                                                                                                                                   
    </html>                                                                                                                                                                   
[2015-04-23 08:59:45.454] [INFO] console - undefined                                                                                                                          
                                                                                                                                                                              ```

the config is the same as website besides the proxy info:

```
    "proxySettings" : {
        "enabled": true,
        "host": "11.103.245.42",
        "port": 8010,
        "tunnel": false
    },
```

**Comment by Hacklone** [comment]
Hey,

It seems like it calls a different public package url than ```http://bower.herokuapp.com/packages```
It should be in the config ```public.registry```
https://github.com/Hacklone/private-bower/blob/master/bower.conf.json

**Comment by igorlino** [comment]
yes that's the one I'm using. Same config file works fine with private-bower 0.6.5. Latest fails.

I'm investigating.

**Comment by Hacklone** [comment]
The format of the config file changed a bit in 1.0.0.
Check out it here: https://github.com/Hacklone/private-bower/blob/master/bower.conf.json

**Comment by igorlino** [comment]
I was already using the config from 1.0.0....  (0.6.5 seems to like it without problems)

In the meantime, I found the problem and a solution. 

The node-rest-client has changed from 1.0.0 to 1.5.0.
I have added some trace code to understand the underlaying HTTP call.
Result: The call to that page looks different.

I made 4 snippets:
1) request in 0.6.5
2) request in 1.0.0
3) new corrected request in 1.0.0 (after my change)
4) Code changes to "publicPackageStore.js" to make the it work.


1) original request 0.6.5
```
>> private-bower --config bower-config.json
[2015-04-23 09:53:23.409] [INFO] console - [bower]  23/4/15 09:53:23   Refreshing public packages...
[2015-04-23 09:53:23.422] [INFO] console - { host: '10.100.255.40',
  port: 8080,
  path: 'http://bower.herokuapp.com/packages/',
  agent: false,
  method: 'GET',
  headers:
   { 'Content-Type': 'application/json',
     'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.90 Safari/537.36',
     host: 'bower.herokuapp.com' } }
[2015-04-23 09:53:23.438] [INFO] console - [bower]  23/4/15 09:53:23   Bower server started on port 5678
[2015-04-23 09:53:28.454] [INFO] console - [bower]  23/4/15 09:53:28   Loaded public packages
```
2) original request 1.0.6
```
>> private-bower --config bower-config.json
[2015-04-23 09:58:06.117] [INFO] console - [bower]  23/4/15 09:58:06   Refreshing public packages...
[2015-04-23 09:58:06.132] [INFO] console - { host: '10.100.255.40',
  port: 8080,
  path: 'http://bower.herokuapp.com/packages/',
  agent: false,
  method: 'GET',
  headers: {} }
[2015-04-23 09:58:06.240] [INFO] console - [bower]  23/4/15 09:58:06   private-bower server started
[2015-04-23 09:58:06.493] [INFO] console - [bower]  23/4/15 09:58:06   Could not load public packages, invalid data!
[2015-04-23 09:58:06.498] [INFO] console - [SyntaxError: Unexpected token <]
```

3)new corrected request in 1.0.0 (after my change)
```
>> private-bower --config bower-config.json
[2015-04-23 09:59:23.965] [INFO] console - [bower]  23/4/15 09:59:23   Refreshing public packages...
[2015-04-23 09:59:23.979] [INFO] console - { host: '10.100.255.40',
  port: 8080,
  path: 'http://bower.herokuapp.com/packages/',
  agent: false,
  method: 'GET',
  headers:
   { 'Content-Type': 'application/json',
     host: 'bower.herokuapp.com',
     'Content-Length': 0 } }
[2015-04-23 09:59:24.075] [INFO] console - [bower]  23/4/15 09:59:24   private-bower server started
[2015-04-23 09:59:30.539] [INFO] console - [bower]  23/4/15 09:59:30   Loaded public packages
[2015-04-23 09:59:30.545] [INFO] console - [bower]  23/4/15 09:59:30   Public registry cached
```

4) code changes
```
  function _loadPublicPackagesPeriodically() {
        var client = createClient();
// set content-type header and data as json in args parameter 
var args = {
  headers:{"Content-Type": "application/json",
  'host': 'bower.herokuapp.com' } 
};
        var req1 = client.get(_publicBowerUrl, args, function(data) {
            processData(data);
        }).on('error', function(err) {
            logger.error('something went wrong on the request', err.request.options);
        });

        client.on('error', function(err) {
            logger.error('Something went wrong on the client', err);
        });
		// view req1 options         
		console.log(req1.options);
```

**Comment by igorlino** [comment]
maybe it  could be a good idea to add a new feature to enable some type of debug/trace log.
```
private-bower --verbose
```

in that mode, then such HTTP request could be displayed. (or other relevant things, to help troubleshooting)

**Comment by igorlino** [comment]
PR at https://github.com/Hacklone/private-bower/pull/137

**Comment by Hacklone** [comment]
Released in version 1.0.7
Thanks everyone :)
